baseapp/abci.go

```diff
@@ -192,8 +192,24 @@ func (app *BaseApp) BeginBlock(req abci.RequestBeginBlock) (res abci.ResponseBeg
 			WithHeaderHash(req.Hash)
 	}
 
+	ctx := app.deliverState.ctx
+	if app.preBeginBlocker != nil {
+		rsp, err := app.preBeginBlocker(ctx, req)
+		if err != nil {
+			panic(fmt.Errorf("preBeginBlock failed, height: %d, err: %w", req.Header.Height, err))
+		}
+		// rsp.ConsensusParamsChanged is true from preBeginBlocker means ConsensusParams in database get changed
```

Comment:
```suggestion
		// rsp.ConsensusParamsChanged is true from preBeginBlocker means ConsensusParams in store get changed
```

store is more accurate, since it's not committed into database yet.